From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HackingGate <i@hackinggate.com>
Date: Fri, 10 Oct 2025 11:51:41 +0900
Subject: [PATCH] arm64: dts: rockchip: rk3576: add driver-required properties

Add missing clock, interrupt, and DMA properties to various device
nodes to enable proper driver functionality:

- usb2phy_grf: Add PCLK_PMUPHY_ROOT clock for USB2 PHY GRF access
- hdptxphy_grf: Add PCLK_PMUPHY_ROOT clock for HDPTX PHY GRF access
- uart1: Add dma-names and interrupts for DMA-based transfers
- vop_mmu: Add interrupt-names and dual clocks for IOMMU operation
- uart10, uart11: Add dma-names and interrupts for serial communication

These properties are required by their respective drivers:
- USB2 PHY driver (phy-rockchip-inno-usb2) needs clocks for bulk
  clock management introduced in v6.13
- UART driver (dw-apb-uart) requires DMA channel names and interrupt
  configuration for efficient data transfers
- VOP IOMMU driver needs both aclk and iface clocks plus named
  interrupts for page fault handling

Without these properties, the affected devices either fail to probe
or operate in degraded mode with missing functionality.

Signed-off-by: HackingGate <i@hackinggate.com>
---
 arch/arm64/boot/dts/rockchip/rk3576.dtsi               | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3576.dtsi b/arch/arm64/boot/dts/rockchip/rk3576.dtsi
index e7e016a..e8106ef 100644
--- a/arch/arm64/boot/dts/rockchip/rk3576.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3576.dtsi
@@ -761,6 +761,7 @@
 			reg = <0x0 0x2602e000 0x0 0x4000>;
 			#address-cells = <1>;
 			#size-cells = <1>;
+			clocks = <&cru PCLK_PMUPHY_ROOT>;
 
 			u2phy0: usb2-phy@0 {
 				compatible = "rockchip,rk3576-usb2phy";
@@ -812,6 +813,7 @@
 		hdptxphy_grf: syscon@26032000 {
 			compatible = "rockchip,rk3576-hdptxphy-grf", "syscon";
 			reg = <0x0 0x26032000 0x0 0x100>;
+			clocks = <&cru PCLK_PMUPHY_ROOT>;
 		};
 
 		vo1_grf: syscon@26036000 {
@@ -878,6 +880,7 @@
 			clocks = <&cru SCLK_UART1>, <&cru PCLK_UART1>;
 			clock-names = "baudclk", "apb_pclk";
 			dmas = <&dmac0 8>, <&dmac0 9>;
+			dma-names = "tx", "rx";
 			interrupts = <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&uart1m0_xfer>;
@@ -1398,6 +1401,7 @@
 			compatible = "rockchip,rk3576-iommu", "rockchip,rk3568-iommu";
 			reg = <0x0 0x27d07e00 0x0 0x100>, <0x0 0x27d07f00 0x0 0x100>;
 			interrupts = <GIC_SPI 342 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "vop_mmu";
 			clocks = <&cru ACLK_VOP>, <&cru HCLK_VOP>;
 			clock-names = "aclk", "iface";
 			#iommu-cells = <0>;
@@ -1457,6 +1461,7 @@
 				     <GIC_SPI 367 IRQ_TYPE_LEVEL_HIGH>;
 			interrupt-names = "avp", "cec", "earc", "main", "hpd";
 			phys = <&hdptxphy>;
+			phy-names = "hdmi";
 			pinctrl-names = "default";
 			pinctrl-0 = <&hdmi_txm0_pins &hdmi_tx_scl &hdmi_tx_sda>;
 			power-domains = <&power RK3576_PD_VO0>;
@@ -2527,6 +2532,7 @@
 			clocks = <&cru SCLK_UART10>, <&cru PCLK_UART10>;
 			clock-names = "baudclk", "apb_pclk";
 			dmas = <&dmac2 21>, <&dmac2 22>;
+			dma-names = "tx", "rx";
 			interrupts = <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&uart10m0_xfer>;
@@ -2541,6 +2547,7 @@
 			clocks = <&cru SCLK_UART11>, <&cru PCLK_UART11>;
 			clock-names = "baudclk", "apb_pclk";
 			dmas = <&dmac2 23>, <&dmac2 24>;
+			dma-names = "tx", "rx";
 			interrupts = <GIC_SPI 87 IRQ_TYPE_LEVEL_HIGH>;
 			pinctrl-names = "default";
 			pinctrl-0 = <&uart11m0_xfer>;
-- 
2.47.3

